libavfilter/dnn/dnn_backend_native_layer_batchnormalization.c

```diff
@@ -0,0 +1,118 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with FFmpeg; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+/**
+ * @file
+ * DNN native backend implementation.
+ */
+
+#include "dnn_backend_native.h"
+#include "libavutil/avassert.h"
+#include "dnn_backend_native_layer_batchnormalization.h"
+
+int ff_dnn_load_layer_batchnormalization(Layer *layer, AVIOContext *model_file_context, int file_size, int operands_num)
+{
+    DnnLayerBatchnormalizationParams *params;
+    int dnn_size = 0;
+    int32_t mean_size = 0;
+    params = av_malloc(sizeof(*params));
+    if (!params)
+        return 0;
+
+    params->channel = avio_rl32(model_file_context);
+    dnn_size += 4;
+
+    mean_size = params->channel;
+    params->mean = av_malloc_array(mean_size, sizeof(*params->mean));
+    if (!params->mean) {
+        av_freep(&params);
+        return 0;
+    }
+
+    params->variance = av_malloc_array(mean_size, sizeof(*params->variance));
+    if (!params->variance) {
+        av_freep(&params->mean);
+        av_freep(&params);
+        return 0;
+    }
+
+    for (int32_t i = 0; i < mean_size; ++i) {
+        params->mean[i] = av_int2float(avio_rl32(model_file_context));
+    }
+    for (int32_t i = 0; i < mean_size; ++i) {
+        params->variance[i] = av_int2float(avio_rl32(model_file_context));
+    }
+    dnn_size += params->channel * 4 * 2;
+
+    params->variance_eps = av_int2float(avio_rl32(model_file_context));
+    params->scale = av_int2float(avio_rl32(model_file_context));
+    params->offset = av_int2float(avio_rl32(model_file_context));
+    dnn_size += 12;
+
+    layer->params = params;
+    layer->input_operand_indexes[0] = (int32_t)avio_rl32(model_file_context);
+    layer->output_operand_index = (int32_t)avio_rl32(model_file_context);
+    dnn_size += 8;
+    if (layer->input_operand_indexes[0] >= operands_num || layer->output_operand_index >= operands_num) {
+        return 0;
+    }
+
+    return dnn_size;
+}
+
+int ff_dnn_execute_layer_batchnormalization(DnnOperand *operands, const int32_t *input_operand_indexes,
+                                            int32_t output_operand_index, const void *parameters, NativeContext *ctx)
+{
+    const DnnOperand *input = &operands[input_operand_indexes[0]];
+    DnnOperand *output = &operands[output_operand_index];
+    const DnnLayerBatchnormalizationParams *params = parameters;
+    int dims_count;
+    const float *src;
+    float *dst;
+    int32_t channel_num = params->channel;
+    const float *mean = params->mean;
+    const float *variance = params->variance;
+    float variance_eps = params->variance_eps;
+    float scale = params->scale;
+    float offset = params->offset;
+
+    for (int i = 0; i < 4; ++i)
+        output->dims[i] = input->dims[i];
```

Comment:
need to check if param->channel matches input->dims[i]

Reply:
ok